src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1513,7 +1543,7 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
                         GetTypeAsDiagnosticArgument(alternativeResult.Type?.TypeSymbol));
                 }
             }
-            resultType = TypeSymbolWithAnnotations.Create(resultType?.TypeSymbol ?? node.Type, isNullableIfReferenceType);
+            resultType = TypeSymbolWithAnnotations.Create(resultType?.TypeSymbol ?? node.Type.SetUnknownNullabilityForReferenceTypes(), isNullableIfReferenceType);
```

Comment:
>SetUnknownNullabilityForReferenceTypes [](start = 94, length = 38)

I thought that for error cases in conditional expressions, we had a rule to pick a winner by default.
Aside from the suppression operator, this is the only use of `SetUnknownNullabilityForReferenceTypes` in the NullableWalker. So feels strange.

Reply:
As we discussed, for a case like `b ? new List<object!>() : new List<object?>()` we should not prefer the nullability from one of the expressions over the other. In that case, the result should be `List<object~>()`.

---
In reply to: [189376456](https://github.com/dotnet/roslyn/pull/26356#discussion_r189376456) [](ancestors = 189376456)